### PR TITLE
feat: groupby aggregation

### PR DESCRIPTION
**Pull request by DarkPhoenix2704**
## Change type

- [x] feat: (new feature for the user, not a new feature for build script)

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/grid/Table.vue [hunk 1]
```diff
@@ -1525,7 +1525,11 @@ watch(
         }
         isViewDataLoading.value = true
         try {
-          await Promise.allSettled([loadData?.(), loadViewAggregate()])
+          if (isGroupBy.value) {
+            await loadData?.()
+          } else {
+            await Promise.allSettled([loadData?.(), loadViewAggregate()])
+          }
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/smartsheet/grid/Table.vue:1532]: errors are handled in the loadData and loadViewAggregate functions

File: packages/nocodb/src/controllers/data-table.controller.ts [hunk 2]
```diff
@@ -132,6 +132,21 @@ export class DataTableController {
...
+    @Req() req: NcRequest,
+    @Param('modelId') modelId: string,
+    @Query('viewId') viewId: string,
+  ) {
+    return await this.dataTableService.bulkAggregate(context, {
+      query: req.query,
+      modelId,
+      viewId,
+    });
+  }
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nocodb/src/controllers/data-table.controller.ts]: they are handled in BaseModelSql2

File: packages/nc-gui/composables/useViewAggregate.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ const [useProvideViewAggregate, useViewAggregate] = useInjectionState(
 
     const aggregations = ref({}) as Ref<Record<string, any>>
 
-    const reloadAggregate = inject(ReloadAggregateHookInj, createEventHook())
+    const reloadAggregate = inject(ReloadAggregateHookInj)
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/composables/useViewAggregate.ts:24]: ignore

File: packages/nc-gui/composables/useViewAggregate.ts [hunk 4]
```diff
@@ -99,26 +99,28 @@
...
-        },
-      ])
+      await reloadAggregate?.trigger({
+        fields: [
+          {
+            title: fields.value.find((f) => f.id === fieldId)?.title ?? '',
+            aggregation: agg,
+          },
+        ],
+      })
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/composables/useViewAggregate.ts:109]: Both are same. Ignore

File: packages/nc-gui/composables/useSharedView.ts [hunk 5]
```diff
@@ -233,6 +233,35 @@ export function useSharedView() {
...
+        aggregateFilterList: JSON.stringify(param.aggregateFilterList),
+        filterArrJson: JSON.stringify(param.filtersArr ?? nestedFilters.value),
+      } as any,
+      {
+        headers: {
+          'xc-password': password.value,
+        },
+      },
+    )
+  }
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/composables/useSharedView.ts:263]: handled in the calling method

File: packages/nocodb/src/db/aggregations/mysql2.ts [hunk 6]
```diff
@@ -58,27 +60,27 @@ export function genMysql2AggregatedQuery({
...
-        aggregationSql = knex.raw(`COUNT(*) AS ??`, [column.id]);
+        aggregationSql = knex.raw(`COUNT(*)`);
         break;
       case CommonAggregations.CountEmpty:
         if ([UITypes.JSON].includes(column.uidt)) {
           aggregationSql = knex.raw(
-            `SUM(CASE WHEN JSON_LENGTH(??) IS NULL THEN 1 ELSE 0 END) AS ??`,
-            [column_query, column.id],
+            `SUM(CASE WHEN JSON_LENGTH(??) IS NULL THEN 1 ELSE 0 END)`,
+            [column_query],
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nocodb/src/db/aggregations/mysql2.ts:77]: The column_query can be a query. So the paramer is required. Ignore this

File: packages/nocodb/src/db/aggregations/sqlite3.ts [hunk 7]
```diff
@@ -58,27 +60,27 @@ export function genSqlite3AggregateQuery({
...
-        aggregationSql = knex.raw(`COUNT(*) AS ??`, [column.id]);
+        aggregationSql = knex.raw(`COUNT(*)`);
         break;
       case CommonAggregations.CountEmpty:
         if ([UITypes.JSON].includes(column.uidt)) {
           aggregationSql = knex.raw(
-            `SUM(CASE WHEN json_array_length(??) IS NULL THEN 1 ELSE 0 END) AS ??`,
-            [column_query, column.id],
+            `SUM(CASE WHEN json_array_length(??) IS NULL THEN 1 ELSE 0 END)`,
+            [column_query],
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nocodb/src/db/aggregations/sqlite3.ts:77]: The column_query can be a query. So the paramer is required. Ignore this

File: packages/nocodb/src/db/aggregations/pg.ts [hunk 8]
```diff
@@ -58,27 +60,26 @@ export function genPgAggregateQuery({
...
         break;
       case CommonAggregations.CountEmpty:
         if ([UITypes.JSON].includes(column.uidt)) {
-          aggregationSql = knex.raw(
-            `COUNT(*) FILTER (WHERE (??) IS NULL) AS ??`,
-            [column_query, column.id],
-          );
+          aggregationSql = knex.raw(`COUNT(*) FILTER (WHERE (??) IS NULL)`, [
+            column_query,
+          ]);
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nocodb/src/db/aggregations/pg.ts:77]: The column_query can be a query. So the paramer is required. Ignore this

File: packages/nocodb/src/db/aggregations/pg.ts [hunk 9]
```diff
@@ -31,12 +32,13 @@
...
     | 'attachment'
     | 'unknown';
+  alias?: string;
 }) {
   let aggregationSql: Knex.Raw | undefined;
 
   const { dbDriver: knex } = baseModelSqlv2;
 
-  let secondaryCondition: any = "''";
+  let condnValue: any = "''";
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nocodb/src/db/aggregations/pg.ts:41]: The do cover. Ignore this

File: packages/nc-gui/components/smartsheet/grid/Aggregation.vue [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<script setup lang="ts">
+defineProps<{
+  group: any
+}>()
```
[the remark this answers is not in the document]
Comment (DarkPhoenix2704) [reply packages/nc-gui/components/smartsheet/grid/Aggregation.vue:4]: Ignore. They are placeholder component. Meant to be like this

**Review by dstala — APPROVED** [review]

**Pull request merged**